provider/pkg/gen/schema.go

```diff
@@ -141,7 +141,16 @@ func PulumiSchema(swagger map[string]interface{}) pschema.PackageSpec {
 						p.name = "t_" + p.name[1:]
 					}
 					objectSpec.Properties[p.name] = genPropertySpec(p, kind.canonicalGV, kind.kind)
-					//objectSpec.Required = append(objectSpec.Required, p.name)
+				}
+
+				if kind.IsNested() {
+					for _, p := range kind.RequiredInputProperties() {
+						objectSpec.Required = append(objectSpec.Required, p.name)
+					}
+				} else {
+					for _, p := range kind.Properties() {
```

Comment:
Do we need to do something similar for `RequiredInputs` on resources, or is that already correct?

Reply:
I believe this is correct. @mikhailshilkov verified that this change produced identical output to the existing .NET SDK.

Reply:
Actually, I'm not so sure. I still see ptr types on some props I don't expect.

Reply:
I double-checked the OpenAPI spec and the client-go types, and it looks like this change is wrong. All of these top-level types are actually marked as optional.

/cc @mikhailshilkov